rewrite-maven/src/main/java/org/openrewrite/maven/tree/MavenRepositoryMirror.java

```diff
@@ -104,7 +104,7 @@ public static MavenRepository apply(Collection<MavenRepositoryMirror> mirrors, M
     }
 
     public MavenRepository apply(MavenRepository repo) {
-        if (repo.getUri().equals(url) && Objects.equals(id, repo.getId()) || !matches(repo)) {
+        if (Objects.equals(id, repo.getId()) && (repo.getUri().equals(url) || repo.isKnownToExist()) || !matches(repo)) {
```

Comment:
I'm having a little trouble following the addition of `repo.isKnownToExist())` here. Could you add some context as to why this is added? This seems to not apply a mirror if a original repository exists, but I don't fully understand why. I'm thinking of scenarios where for instance the mirror limits outbound traffic, where both might exist, but the mirror is still preferred.

It might just be my unfamiliarity with the classes here; from all your earlier work I've come to trust your judgement; just want to be sure I understand why, and that the changes are as intended.

Reply:
`knownToExist` is a flag to test if the repo URL a) had any property substituted; and b) is reachable, in case of a HTTP[S] remote repository. Looks like this was used as some sort of cache shortcut mechanism, but there's also a cache in place for it. This addition avoids any further testing if this repo was tested before.

A few lines below this change, `knownToExist` is set to `false` in case of the URL is replaced by a mirror, and this is the only place this field was set before this change.